coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -144,9 +144,9 @@ class SingleClusterPlanner(val dataset: Dataset,
             reduceExec.addRangeVectorTransformer(new LabelCardinalityPresenter())
             reduceExec
           }
-          case lce: TopkCardExec => {
-            val reducer = TopkCardReduceExec(qContext, targetActor, many, lce.k)
-            reducer.addRangeVectorTransformer(TopkCardPresenter(lce.k))
+          case lce: TsCardExec => {
+            val reducer = TsCardReduceExec(qContext, targetActor, many)
+            reducer.addRangeVectorTransformer(TsCardPresenter(lce.groupDepth))
```

Comment:
Add presenter only if no further reduction is necessary, we will need it when we implement handling multi partition calls (previous comment).

Reply:
Updated